packages/api/src/platforms/vtex/resolvers/product.ts

```diff
@@ -74,13 +78,38 @@ export const StoreProduct: Record<string, Resolver<Root>> & {
       numberOfItems: categories.length,
     }
   },
-  image: ({ images }) =>
-    (nonEmptyArray(images) ?? [DEFAULT_IMAGE]).map(
-      ({ imageUrl, imageText }) => ({
+  image: ({ images }, args) => {
+    const resolvedImages = (nonEmptyArray(images) ?? [DEFAULT_IMAGE]).map(
+      ({ imageUrl, imageText, imageLabel }) => ({
         alternateName: imageText ?? '',
         url: imageUrl.replace('vteximg.com.br', 'vtexassets.com'),
+        keywords: imageLabel,
       })
-    ),
+    )
+
+    if(typeof args !== 'object') {
+      return resolvedImages;
+    }
+
+    let { context, limit } = args as StoreProductImageArgs
+
+    const shouldFilter = context !== "generic"
+
+    // Normalize count to undefined as we want any negative value to always return the full list of images
+    limit = limit || -1
+    limit = limit <= -1 ? undefined : limit
```

Comment:
`limit = 0` should be transformed to `-1`?

otherwise I'd suggest 
```suggestion
    // Normalize count to undefined as we want any negative value to always return the full list of images
    limit = limit < 0 ? undefined : limit;
```

Reply:
The issue is that the limit can be undefined or null, but javascript never ceases to amaze me `(undefined < 0) === false`